src/protocols/secure_channel/SessionEstablishmentDelegate.h

```diff
@@ -25,13 +25,27 @@
 
 #pragma once
 
+#include <protocols/secure_channel/PairingSession.h>
 #include <system/SystemPacketBuffer.h>
 #include <transport/Session.h>
 #include <transport/raw/MessageHeader.h>
 #include <transport/raw/PeerAddress.h>
 
 namespace chip {
 
+enum class SessionState : uint8_t
+{
+    kUndefined              = 0,
+    kSentPBKDFParamRequest  = 1,
```

Comment:
I think these values could use some clear documentation explaining that these describe the stage of the overall key exchange, not the step we, as initiator, have gotten to.  Because it's pretty confusing to have kSentSigma2 showing up on the initiator side as thing stand.

Alternately, we could be more fine-grained in terms of kSentSigma1, kReceivedSigma1, kSentSigma2, kReceivedSigma2, etc.  That will incidentally let us detect errors in between "received sigma1" and "sent sigma2" and similar between "received sigma2" and "sent sigma3".

Because as things stand, both "we sent sigma1 and then got no response" and "we sent sigma1, got back a sigma2, but failed to process it" are being maped to kSentSigma1.

This would require adding some states to the state machine in CASESession, but that should be pretty doable, I think...  Though probably it should happen as a followup to this PR.

Also, it looks like we are not implementing the PASE values here for now, right?  Is the plan to do them as part of this PR?  If not, I would just remove them from the enum for now so people are clear on what they can expect for the moment....